mflowgen/glb_top/custom-init/outputs/pin-assignments.tcl

```diff
@@ -93,20 +99,30 @@ for {set j 0} {$j < $array_width} {incr j} {
     set tile_ports($j) [lsort -command port_compare $tile_ports($j)]
 }
 
-set left [lsort -command port_compare $left]
-
+set first_tile [lsort -command port_compare $first_tile]
 set width [dbGet top.fPlan.box_urx]
 set height [dbGet top.fPlan.box_ury]
+set first_tile_pin_range [expr {[llength $first_tile] * 0.5 + 5}]
+set center [expr {($width - $first_tile_pin_range) / 2 + $first_tile_pin_range}]
 set tile_width [dbGet [dbGet -p top.insts.name *glb_tile* -i 0].cell.size_x]
+set left_offset 20
 
-editPin -pin $left -start { 0 10 } -end [list 0 [expr {$height - 5}]] -side LEFT -spreadType RANGE -spreadDirection clockwise -layer M6
-editPin -pin "clk" -assign {0 5} -layer M6
+set glb_tile_port_offset [expr {($width - $first_tile_pin_range) / $num_glb_tiles }]
+ 
+# pins for glb <-> soc
+editPin -pin $first_tile -start 5 $height -end $first_tile_pin_range $height -side TOP -spreadType RANGE -spreadDirection clockwise -layer M5
 
+# clk pin
+editPin -pin "clk" -side TOP -assign [expr $center-10]  $height -layer M5
+editPin -pin "reset" -side TOP -assign [expr $center+10] $height -layer M5
+
+# for glb <-> glc
 for {set i 0} {$i < $num_glb_tiles} {incr i} {
-    editPin -pin $glb_tile_ports($i) -start [list [expr {$tile_width*$i+5}] 0] -end [list [expr {$tile_width*$i+8}] 0] -side BOTTOM -spreadType RANGE -spreadDirection counterclockwise -layer M5
+    editPin -pin $glb_tile_ports($i) -start [list [expr {$first_tile_pin_range + $glb_tile_port_offset * $i + 20}] $height] -end [list [expr {$first_tile_pin_range + $glb_tile_port_offset * ($i + 1) - 20}] $height] -side TOP -spreadType RANGE -spreadDirection clockwise -layer M5
```

Comment:
So from this, we should place the GLC and the SoC SRAMs at the top left of the GLB in the top-level floorplan?

Reply:
> So from this, we should place the GLC and the SoC SRAMs at the top left of the GLB in the top-level floorplan?

Yes, if `glc` is placed at top-left of `glb` it will be good to minimize the top-level routing congestion.
`glb` communicates with `SoC` axi-interconnect, so I am not sure if placing SoC SRAMs at the top left would be necessary. Placing SoC logics at the top of glb would be preferred for the same reason as `glc`.